test/jit/test_torchbind.py

```diff
@@ -327,6 +327,22 @@ def forward(self):
         traced = torch.jit.trace(TryTracing(), ())
         self.assertEqual(torch.zeros(4, 4), traced())
 
+    def test_torchbind_overloaded_method(self):
+        def f():
+            val = torch.classes._TorchScriptTesting._FooOverload(3, 5)
+            stuff1 = val.increment(4, 5)
+            stuff2 = val.increment(4)
+            stuff3 = val.increment(4, "hi")
+            return stuff1, stuff2, stuff3
+        self.checkScript(f, ())
+
+    def test_torchbind_overloaded_method_init(self):
+        def f():
+            val = torch.classes._TorchScriptTesting._FooOverloadInit(3, 4, "hi")
```

Comment:
Can you test overloaded init that has slightly different behavior, so that we can be sure the right method is called? Also I would recommend calling both init variants here to make sure they both work.